development/build/transforms/remove-fenced-code.test.js

```diff
@@ -0,0 +1,564 @@
+
+    // Valid inputs
+    Object.keys(BuildTypes).forEach((buildType) => {
+      it(`transforms file with fences for build type "${buildType}"`, () => {
+        expect(
+          removeFencedCode(
+            mockFileName,
+            buildType,
+            testData.validInputs.withFences,
+          ),
+        ).toStrictEqual(testData.validOutputs[buildType]);
+
+        // Ensure that the minimal input template is in fact valid
+        const minimalInput = getMinimalFencedCode(buildType);
+        expect(
+          removeFencedCode(mockFileName, buildType, minimalInput),
+        ).toStrictEqual([minimalInput, false]);
+      });
+
+      it(`does not modify file without fences for build type "${buildType}"`, () => {
+        expect(
+          removeFencedCode(
+            mockFileName,
+            buildType,
+            testData.validInputs.withoutFences,
+          ),
+        ).toStrictEqual([testData.validInputs.withoutFences, false]);
+      });
+    });
+
+    // Invalid inputs
+    it('rejects empty fences', () => {
+      const jsComment = '// A comment\n';
+
+      const emptyFence = getMinimalFencedCode()
+        .split('\n')
+        .filter((line) => line.startsWith('///:'))
+        .map((line) => `${line}\n`)
+        .join('');
+
+      const emptyFenceWithPrefix = jsComment.concat(emptyFence);
+      const emptyFenceWithSuffix = emptyFence.concat(jsComment);
+      const emptyFenceSurrounded = emptyFenceWithPrefix.concat(jsComment);
+
+      const inputs = [
+        emptyFence,
+        emptyFenceWithPrefix,
+        emptyFenceWithSuffix,
+        emptyFenceSurrounded,
+      ];
+
+      inputs.forEach((input) => {
+        expect(() =>
+          removeFencedCode(mockFileName, BuildTypes.flask, input),
+        ).toThrow(
+          `Empty fence found in file "${mockFileName}":\n${emptyFence}`,
+        );
+      });
+    });
+
+    it('rejects sentinels preceded by non-whitespace', () => {
+      // Matches the sentinel component of the first line beginning with "///:"
+      const fenceSentinelRegex = /^\/\/\/:/mu;
+      const replacements = ['a ///:', '2 ///:', '_ ///:'];
+
+      replacements.forEach((replacement) => {
+        expect(() =>
+          removeFencedCode(
+            mockFileName,
+            BuildTypes.flask,
+            getMinimalFencedCode().replace(fenceSentinelRegex, replacement),
+          ),
+        ).toThrow(
+          /Fence sentinel may only appear at the start of a line, optionally preceded by whitespace.$/u,
+        );
+      });
+    });
+
+    it('rejects sentinels not followed by a single space and a multi-character alphabetical string', () => {
+      // Matches the sentinel and terminus component of the first line
+      // beginning with "///: TERMINUS"
+      const fenceSentinelAndTerminusRegex = /^\/\/\/: \w+/mu;
+
+      const replacements = [
+        '///:BEGIN',
+        '///:XBEGIN',
+        '///:_BEGIN',
+        '///:B',
+        '///:_',
+        '///: ',
```

Comment:
Nit: It would be nice to include a single-character alphabetical test case here. The description isn't met without it.
```suggestion
        '///: ',
        '///: B',
```

Reply:
474ec02